sipi/scripts/convert_from_path.lua

```diff
@@ -48,113 +46,106 @@ end
+    send_error(500, "server.fs.is_readable() failed: " .. readable)
     return
 end
 
 if not readable then
-
-    send_error(500, FILE_NOT_READABLE .. sourcePath)
+    send_error(400, FILE_NOT_READABLE .. sourcePath)
     return
 end
 
 -- check for the mimetype of the file
-success, real_mimetype = server.file_mimetype(sourcePath)
+local mime_info
+success, mime_info = server.file_mimetype(sourcePath)
 
 if not success then
-    server.log("server.file_mimetype() failed: " .. exists, server.loglevel.LOG_ERR)
-    send_error(500, "mimetype of file could not be determined")
+    send_error(500, "server.file_mimetype() failed: " .. mime_info)
+    return
 end
 
+local mime_type = mime_info["mimetype"]
+
 -- handle the file depending on its media type (image, text file)
-mediatype = get_mediatype(real_mimetype.mimetype)
+local file_info = get_file_info(originalFilename, mime_type)
 
 -- in case of an unsupported mimetype, the function returns false
-if not mediatype then
-    send_error(400, "Mimetype '" .. real_mimetype.mimetype .. "' is not supported")
+if not file_info then
+    send_error(400, "Mimetype '" .. mime_type .. "' is not supported")
     return
 end
 
+local media_type = file_info["media_type"]
+
 -- depending on the media type, decide what to do
-if mediatype == IMAGE then
+if media_type == IMAGE then
 
     -- it is an image
 
     --
     -- check if project directory is available, if not, create it
     --
 
-    projectDir = config.imgroot .. '/' .. prefix .. '/'
+    local projectDir = config.imgroot .. '/' .. prefix .. '/'
 
+    local exists
     success, exists = server.fs.exists(projectDir)
     if not success then
-        server.log("server.fs.exists() failed: " .. exists, server.loglevel.LOG_ERR)
+        send_error(500, "server.fs.exists() failed: " .. exists)
+        return
     end
 
     if not exists then
         success, errmsg = server.fs.mkdir(projectDir, 511)
         if not success then
-            server.log("server.fs.mkdir() failed: " .. errmsg, server.loglevel.LOG_ERR)
-            send_error(500, "Project directory could not be created on server")
+            send_error(500, "server.fs.mkdir() failed: " .. errmsg)
             return
         end
     end
 
+    local baseName
     success, baseName = server.uuid62()
     if not success then
-        server.log("server.uuid62() failed: " .. baseName, server.loglevel.LOG_ERR)
-        send_error(500, "unique name could not be created")
+        send_error(500, "server.uuid62() failed: " .. baseName)
         return
     end
 
 
     --
     -- create full quality image (jp2)
     --
+    local fullImg
     success, fullImg = SipiImage.new(sourcePath)
     if not success then
-        server.log("SipiImage.new() failed: " .. fullImg, server.loglevel.LOG_ERR)
+        send_error(500, "SipiImage.new() failed: " .. fullImg)
         return
     end
 
-    --success, check = fullImg:mimetype_consistency(originalMimeType, originalFilename)
```

Comment:
Where does Sipi check the actual mime type of the file?

Reply:
It doesn't in this Lua script; this code was already commented out, but I don't know why.

Reply:
I think it should be commented in again. Otherwise, the check is not performed.

Reply:
Then the integration test won't work anymore, I think. I commented it out. You are of course welcome to find a solution for the test that allows for the mime-type check to be added :-)

Reply:
I think there will also be situations where a problem cannot be fixed at once. But we should keep track of such problems by creating issues.